services/app-web/src/components/ChangeHistory/ChangeHistory.test.tsx

```diff
@@ -104,13 +149,60 @@ describe('Change History', () => {
         ).not.toBeVisible()
         const accordionRows = screen.getAllByRole('button')
         expect(accordionRows[0]).toHaveTextContent(
-            '03/23/22 9:19pm ET - Submission'
+            '03/24/22 9:19pm ET - Submission'
         )
         expect(accordionRows[1]).toHaveTextContent(
-            '03/23/22 9:18pm ET - Unlock'
+            '03/24/22 9:18pm ET - Unlock'
         )
         expect(accordionRows[2]).toHaveTextContent(
+            '03/23/22 9:19pm ET - Submission'
+        )
+        expect(accordionRows[3]).toHaveTextContent(
+            '03/23/22 9:18pm ET - Unlock'
+        )
+        expect(accordionRows[4]).toHaveTextContent(
             '03/22/22 10:08pm ET - Submission'
         )
     })
+    it('has correct href values for previous submission links', () => {
+        render(<ChangeHistory submission={submissionData} />)
+        expect(screen.getByTestId(`revision-link-0`)).toHaveAttribute(
+            'href',
+            `/submissions/440d6a53-bb0a-49ae-9a9c-da7c5352789f/revisions/0`
+        )
+        expect(screen.getByTestId(`revision-link-1`)).toHaveAttribute(
+            'href',
+            `/submissions/440d6a53-bb0a-49ae-9a9c-da7c5352789f/revisions/1`
+        )
+    })
+    it('should list accordion items with links when appropriate', () => {
+        render(<ChangeHistory submission={submissionData} />)
+        //Latest resubmission should not have a link.
+        expect(
+            screen.getByTestId('accordionItem_2022-03-25T01:19:46.154Z')
+        ).not.toHaveTextContent('View past submission version')
+        //Unlock history accordion should not have a link
+        expect(
+            screen.getByTestId('accordionItem_2022-03-25T01:18:44.663Z')
+        ).not.toHaveTextContent('View past submission version')
+        //Previous submission should contain a link
+        expect(
+            screen.getByTestId('accordionItem_2022-03-24T01:19:46.154Z')
+        ).toHaveTextContent('View past submission version')
+        //Unlock history accordion should not have a link
+        expect(
+            screen.getByTestId('accordionItem_2022-03-24T01:18:44.663Z')
+        ).not.toHaveTextContent('View past submission version')
+        //Previous submission should contain a link
+        expect(
+            screen.getByTestId('accordionItem_2022-03-23T02:08:52.259Z')
+        ).toHaveTextContent('View past submission version')
+    })
+    it('should not list links for initial submission without revisions', () => {
+        render(<ChangeHistory submission={submissionDataInitialSubmission} />)
+        //Initial submission should not have a link
+        expect(
+            screen.getByTestId('accordionItem_2022-03-23T02:08:52.259Z')
```

Comment:
Thank you for these tests they are great 🌮  also going to help us stabilize these changes.